clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -1872,6 +1882,201 @@ check_timestamps_increase_monotonically(void)
+        instrlist_t *ilist = instrlist_create(GLOBAL_DCONTEXT);
+        instrlist_append(ilist, store);
+        instrlist_append(ilist, nop);
+
+        std::vector<memref_with_IR_t> memref_instr_vec = {
+            { gen_marker(TID, TRACE_MARKER_TYPE_FILETYPE, OFFLINE_FILE_TYPE_ENCODINGS),
+              nullptr },
+            { gen_instr(TID), store },
+            { gen_instr(TID), nop },
+        };
+
+        static constexpr addr_t BASE_ADDR = 0xeba4ad4;
+        auto memrefs = add_encodings_to_memrefs(ilist, memref_instr_vec, BASE_ADDR);
+        instrlist_clear_and_destroy(GLOBAL_DCONTEXT, ilist);
+        if (!run_checker(memrefs, true,
+                         { "Missing write records", TID, /*ref_ordinal=*/3,
+                           /*last_timestamp=*/0,
+                           /*instrs_since_last_timestamp=*/2 },
+                         "Fail to catch missing write records"))
+            return false;
+    }
+#if defined(X86_64) || defined(X86_32)
+    // Correct: ignore predicated operands which may not have memory access.
+    {
+        instr_t *nop = XINST_CREATE_nop(GLOBAL_DCONTEXT);
+        instr_t *rep_movs = INSTR_CREATE_rep_movs_1(GLOBAL_DCONTEXT);
+        instrlist_t *ilist = instrlist_create(GLOBAL_DCONTEXT);
+        instrlist_append(ilist, rep_movs);
+        instrlist_append(ilist, nop);
+        static constexpr addr_t BASE_ADDR = 0xeba4ad4;
+        std::vector<memref_with_IR_t> memref_setup = {
+            { gen_marker(TID, TRACE_MARKER_TYPE_FILETYPE, OFFLINE_FILE_TYPE_ENCODINGS),
+              nullptr },
+            { gen_instr(TID), rep_movs },
+            { gen_instr(TID), nop },
+        };
+        auto memrefs = add_encodings_to_memrefs(ilist, memref_setup, BASE_ADDR);
+        instrlist_clear_and_destroy(GLOBAL_DCONTEXT, ilist);
+        if (!run_checker(memrefs, false)) {
+            return false;
+        }
+    }
+    // Correct: ignore operands with opcode which do not have real memory
+    // access.
+    {
+        instr_t *lea =
+            INSTR_CREATE_lea(GLOBAL_DCONTEXT, opnd_create_reg(REG1),
+                             opnd_create_base_disp(REG1, REG_NULL, 0, 1, OPSZ_lea));
+        instr_t *nop = XINST_CREATE_nop(GLOBAL_DCONTEXT);
+        instrlist_t *ilist = instrlist_create(GLOBAL_DCONTEXT);
+        instrlist_append(ilist, lea);
+        instrlist_append(ilist, nop);
+        static constexpr addr_t BASE_ADDR = 0xeba4ad4;
+        std::vector<memref_with_IR_t> memref_setup = {
+            { gen_marker(TID, TRACE_MARKER_TYPE_FILETYPE, OFFLINE_FILE_TYPE_ENCODINGS),
+              nullptr },
+            { gen_instr(TID), lea },
+            { gen_instr(TID), nop },
+        };
+        auto memrefs = add_encodings_to_memrefs(ilist, memref_setup, BASE_ADDR);
+        instrlist_clear_and_destroy(GLOBAL_DCONTEXT, ilist);
+        if (!run_checker(memrefs, false)) {
+            return false;
+        }
+    }
```

Comment:
Please add at least one test of cache flush instructions (see type_is_read() comment).

Reply:
done.